src/Mvc/test/Mvc.IntegrationTests/AuthorizeFilterIntegrationTest.cs

```diff
@@ -72,18 +140,18 @@ private static ApplicationModelProviderContext GetProviderContext(Type controlle
             return context;
         }
 
-        private static IServiceProvider GetServices()
+        private static IServiceProvider GetServices(bool combineAuthorize = false)
         {
             var serviceCollection = new ServiceCollection();
             serviceCollection.AddAuthorization();
-            serviceCollection.AddMvc();
+            serviceCollection.AddMvc(o => o.AllowCombiningAuthorizeFilters = combineAuthorize);
             serviceCollection
                 .AddTransient<ILoggerFactory, LoggerFactory>()
                 .AddTransient<ILogger<DefaultAuthorizationService>, Logger<DefaultAuthorizationService>>()
                 .AddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
             return serviceCollection.BuildServiceProvider();
         }
-
+        
```

Comment:
Functional test with the repro for the bug that's being fixed?

Reply:
I will try to port the new functional tests in 3.0 that pranav added for this scenario to 2.1/2.2 https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/test/Mvc.FunctionalTests/AuthMiddlewareAndFilterTestBase.cs